src/tests/string-case.ts

```diff
@@ -0,0 +1,73 @@
+import { doNotExecute, Expect, NotEqual, Equal } from "./utils";
+
+doNotExecute(async () => {
+  const string: "A" | "B" = "A";
+
+  const lowerCaseString = string.toLowerCase();
+
+  type tests = [Expect<NotEqual<typeof string, typeof lowerCaseString>>];
+});
+
+doNotExecute(async () => {
+  const string: "A" | "B" = "A";
+
+  const lowerCaseString = string.toLowerCase();
+
+  type tests = [
+    Expect<Equal<Lowercase<typeof string>, typeof lowerCaseString>>
```

Comment:
it would be much better if it specifies an explicit expected result
```suggestion
    Expect<Equal<typeof lowerCaseString, "a" | "b">>
```